### PR TITLE
fix(core,query): handle edge cases with writeAtom returning a promise

### PR DESCRIPTION
**Pull request by dai-shi**
#689 changed `writeAtom` to return a promise. This is a follow-up PR to cover some edge cases.

- remove extra code in code that is basically unnecessary in jotai core
- make atomWithQuery write function to return a promise with fixing tests in jotai/query

### CODE REVIEW

File: src/core/store.ts [hunk 1]
```diff
@@ -365,11 +363,7 @@ export const createStore = (
   ): void | Promise<void> => {
     const writePromise = getAtomState(atom)?.w
     if (writePromise) {
-      writePromise.then(() => {
-        writeAtomState(atom, update)
-        flushPending()
-      })
-      return
+      return writePromise.then(() => writeAtomState(atom, update))
```
[the remark this answers is not in the document]
Comment (Thisen) [reply src/core/store.ts:366]: What I meant, is if you can write a test inside core, for the edge cases - Not the particular line. I'm sorry for not being clear.

[the remark this answers is not in the document]
Comment (dai-shi) [reply src/core/store.ts:366]: Sorry, I wasn't clear. You made a good point, we don't have a test for this, even if you don't know what it does.
The reason we didn't have a test is because we didn't know the actual use case.
Now, I feel like this isn't necessary at all. I will remove it completely.

File: src/core/store.ts [hunk 2]
```diff
@@ -196,12 +196,13 @@ export const createStore = (
...
     atom: Atom<Value>,
-    promise?: Promise<void>
+    promise: Promise<void> | null,
+    prevPromise?: Promise<void>
   ): void => {
     const [atomState] = wipAtomState(atom)
     if (promise) {
       atomState.w = promise
-    } else {
+    } else if (atomState.w === prevPromise) {
```
Comment (Thisen) [inline src/core/store.ts:205]: Could you explain this with a comment?

**Review by Thisen — APPROVED** [review]
LGTM, only one nitpick request. :-)
